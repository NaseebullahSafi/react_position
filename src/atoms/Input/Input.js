import React from 'react'
import './Input.scss'

/**
 * Input component.
 * @param {Object} props 
 */
const Input = props => <input className="input" {...props} />

export default Input;