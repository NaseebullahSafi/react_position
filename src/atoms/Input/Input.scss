.input {
  background-color: #4d596114;
  border: none;
  border-radius: 5px;
  font-weight: bold;
  letter-spacing: 3px;
  margin: 0 11px;
  outline: none;
  padding: 6px;
  text-align: center;
  transition: all 0.25s ease-in-out;
  width: 58px;

  &:focus,
  &:active {
    background-color: #2bc2a3;
    box-shadow: 0 0 4px -2px #4d5961;
    transform: scale(1.08);
  }
}
