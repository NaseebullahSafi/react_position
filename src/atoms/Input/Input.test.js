import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Input from './Input';

afterEach(cleanup)

describe('Input Component', () => {
  test('Matches the snapshot', () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot()
  })
})