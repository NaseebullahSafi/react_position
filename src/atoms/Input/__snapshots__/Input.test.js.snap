// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Matches the snapshot 1`] = `
<DocumentFragment>
  <input
    class="input"
  />
</DocumentFragment>
`;
